sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -99,13 +99,20 @@ class SymmetricHashJoinStateManager(
   /**
    * Get all the matched values for given join condition, with marking matched.
    * This method is designed to mark joined rows properly without exposing internal index of row.
+   *
+   * @param joinOnlyFirstTimeMatchedRow Only join with first-time matched row.
```

Comment:
`skipMatchedRowsInPreviousBatches`?

Reply:
Or like `excludeRowsAlreadyMatched`. I guess we'd like to have another method to deal with left-semi join efficiently like I commented (and in the PR description as well).

Reply:
IMO it would be good to add the early eviction for left side matched row, in a follow up PR. If people strongly think we should add that at the first place with this PR, I can add as well, but need more time to polish it.

Reply:
That would depend on whether the follow PR would revert the change done here (say, back and forth) or not. If it's likely to do the back-and-forth and the follow-up is not a big thing (like less than 100 lines of code change and could be done in a couple of days), then probably better to address here to save efforts on reviewing something which will be reverted later.

Reply:
I'm OK to leave it as a follow up.